src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -979,450 +960,191 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                     Else
-                        Return MakeAsteriskToken(precedingTrivia, False)
+                        Return MakeAsteriskToken(precedingTrivia, fullWidth)
                     End If
-
                 Case "/"c
-                    Dim lengthWithMaybeEquals = 1
                     If TrySkipFollowingEquals(lengthWithMaybeEquals) Then
                         Return MakeSlashEqualsToken(precedingTrivia, lengthWithMaybeEquals)
                     Else
-                        Return MakeSlashToken(precedingTrivia, False)
+                        Return MakeSlashToken(precedingTrivia, fullWidth)
                     End If
-
                 Case "\"c
-                    Dim lengthWithMaybeEquals = 1
                     If TrySkipFollowingEquals(lengthWithMaybeEquals) Then
                         Return MakeBackSlashEqualsToken(precedingTrivia, lengthWithMaybeEquals)
                     Else
-                        Return MakeBackslashToken(precedingTrivia, False)
+                        Return MakeBackslashToken(precedingTrivia, fullWidth)
                     End If
-
                 Case "^"c
-                    Dim lengthWithMaybeEquals = 1
                     If TrySkipFollowingEquals(lengthWithMaybeEquals) Then
                         Return MakeCaretEqualsToken(precedingTrivia, lengthWithMaybeEquals)
                     Else
-                        Return MakeCaretToken(precedingTrivia, False)
+                        Return MakeCaretToken(precedingTrivia, fullWidth)
                     End If
-
                 Case "!"c
-                    Return MakeExclamationToken(precedingTrivia, False)
-
+                    Return MakeExclamationToken(precedingTrivia, fullWidth)
                 Case "."c
                     If CanGet(1) AndAlso IsDecimalDigit(Peek(1)) Then
                         Return ScanNumericLiteral(precedingTrivia)
                     Else
-                        Return MakeDotToken(precedingTrivia, False)
+                        Return MakeDotToken(precedingTrivia, fullWidth)
                     End If
-
-                Case "0"c,
-                      "1"c,
-                      "2"c,
-                      "3"c,
-                      "4"c,
-                      "5"c,
-                      "6"c,
-                      "7"c,
-                      "8"c,
-                      "9"c
+                Case "0"c, "1"c, "2"c, "3"c, "4"c,
+                     "5"c, "6"c, "7"c, "8"c, "9"c
                     Return ScanNumericLiteral(precedingTrivia)
-
                 Case """"c
                     Return ScanStringLiteral(precedingTrivia)
-
                 Case "A"c
                     If NextAre(1, "s ") Then
-
                         ' TODO: do we allow widechars in keywords?
                         Dim spelling = "As"
-                        AdvanceChar(2)
+                        If fullWidth Then spelling = GetText(2) Else AdvanceChar(2)
```

Comment:
I've had the feeling that this could be simplified to
```
Return MakeKeyword(SyntaxKind.AsKeyword, "If", precedingTrivia)
```
but never entirely sure.